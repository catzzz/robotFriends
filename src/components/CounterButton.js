import React, { Component } from "react";

class CounterButton extends Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,

    };
    this.udateCount = this.udateCount.bind(this);
  }
  shouldComponentUpdate(nextProps, nextState) {
    // console.log(nextProps, nextState);
    // return true;
    if (this.state.count !== nextState.count) {
      return true;
    }
    return false;
  }
  udateCount() {
    this.setState((state) => {
      return { count: state.count + 1 };
    });
  }
  render() {
    console.log("CounterButton render");
    return (
      <button
        id="counter"
        color={this.props.color}
        onClick={() => this.setState((state) => ({ count: state.count + 1 }))}
      >
        Count: {this.state.count}
      </button>
    );
  }
}

export default CounterButton;
