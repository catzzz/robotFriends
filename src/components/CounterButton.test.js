import CounterButton from './CounterButton';
import { create } from "react-test-renderer";
import React from 'react';
import {shallow} from 'enzyme'

it('expect to render CounterButton componet ',()=>{
    const mockColor = 'red';
    const app = create(<CounterButton color = {mockColor}/>);
    expect(app.toJSON()).toMatchSnapshot();
})

// it('correctly increment the counter',()=>{
//     const mockColor = 'red';
    
//     const wrapper = shallow(<CounterButton color = {mockColor}/>);
//     wrapper.find('[id="counter"]').simulate('click');
//     wrapper.find('[id="counter"]').simulate('click');
//     expect(wrapper.state()).toEqual({count:2});
// })
it('correnctly increments the counter', () => {
    const mockColor = 'red';
    const wrapper = shallow(<CounterButton color= {mockColor}/>)

    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 1})
    wrapper.find('[id="counter"]').simulate('click');
    wrapper.find('[id="counter"]').simulate('click');
    expect(wrapper.state()).toEqual({count: 3})
    expect(wrapper.props().color).toEqual("red")
  });