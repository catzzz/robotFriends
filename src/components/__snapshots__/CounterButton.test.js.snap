// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correnctly increments the counter 1`] = `ShallowWrapper {}`;

exports[`expect to render CounterButton componet  1`] = `
<button
  color="red"
  id="counter"
  onClick={[Function]}
>
  Count: 
  0
</button>
`;
